rtos/source/ThisThread.cpp

```diff
@@ -251,4 +251,32 @@ const char *get_name()
 #endif
 }
 
+osPriority_t ThisThread::get_priority()
+{
+#if MBED_CONF_RTOS_PRESENT
+    osThreadId_t id = osThreadGetId();
+    if (id == nullptr) {
+        return osPriorityError;
+    }
+
+    return static_cast<osPriority_t>(osThreadGetPriority(id));
+#else
+    return static_cast<osPriority_t>(osPriorityError);
```

Comment:
Should we error here or do something more sensible for applications instead?

Reply:
Seems sensible to return "normal" for non-RTOS. We basically view it as a single-thread system, and may as well say the only thread is normal priority.

But, as with the other API addition - what's the use case?

There's an issue that OSes can be a bit funny about their main thread sometimes, so it doesn't necessarily make sense to have every single `Thread` API in `ThisThread` - implementation for `main` could be problematic. It happens not to be in RTX, but I remain a tad wary. (I've been following the lead of C++11 here somewhat).

Reply:
I agree with Kevin, it's not really an error.